### DISCUSSION TITLE
vim.lsp.buf.hover question

### DISCUSSION THREAD

**Discussion by jkopano** [neovim/neovim#28165] [General]
Hi,
So  i noticed that vim.lsp.buf.hover() doesnt show parameters of class construct when inicializing new object in cpp, so the question is can i change this behavior somehow or are there any plugins that cover this subject?
![2024-04-03_19-33](https://github.com/neovim/neovim/assets/62995382/e5a23ad4-5847-4e73-9e34-be0cd8689f82)

**Comment by zeertzjq** [comment]
You can get it by running `vim.lsp.buf.hover()` on the `v` variable name or running `vim.lsp.buf.signature_help()` between the parentheses.

**Comment by jkopano** [comment]
The closest to the thing i want to achieve is running `vim.lsp.buf.signature()` on one of the parameters of contructor, but i want to get this output:
![image](https://github.com/neovim/neovim/assets/62995382/f4e2d868-8a95-490a-9bc3-12c858365ad9)
without the need to write these parameters, which does this:
![image](https://github.com/neovim/neovim/assets/62995382/f7032f85-757d-4374-9425-4305fa5c7b19)
btw running `vim.lsp.buf.signature_help()` on first argument of this constructor is returning the same thing
![image](https://github.com/neovim/neovim/assets/62995382/ba1ef6a5-2280-4e40-9587-5931378462e8)

To be clear with what i want to achieve -  i just want to know what arguments does contructor take without needing to go to its definition.

**Comment by jkopano** [comment]
[follow up](https://www.reddit.com/r/neovim/comments/1buy8yk/comment/ky1v7kf/?context=3) if someone in future gets the same issue

**Reply by mxhuntr** [reply]
### How to get the round border in neovim 0.12+ 
```
vim.lsp.handlers.hover, {
        border = "rounded",
    }
```
### This is deprecated.
